packages/financial-templates-lib/src/price-feed/PerlinXPriceFeed.js

```
@@ -0,0 +1,291 @@
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   * @param {Bool} invertPrice Indicates if prices should be inverted before returned.
+   * @param {Number} priceFeedDecimals Number of priceFeedDecimals to use to convert price to wei.
+   * @param {Number} periods Number of minutes interval between price entries.
+   */
+  constructor(
+    logger,
+    web3,
+    tradermadeApiKey,
+    cryptowatchApiKey,
+    pair,
+    convertToPerl,
+    lookback,
+    networker,
+    getTime,
+    minTimeBetweenUpdates,
+    invertPrice,
+    priceFeedDecimals = 18,
+    periods = 1
+  ) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.tradermadeApiKey = tradermadeApiKey;
+    this.cryptowatchApiKey = cryptowatchApiKey;
+    this.pair = pair;
+    this.convertToPerl = convertToPerl;
+    this.lookback = lookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.invertPrice = invertPrice;
+    this.priceFeedDecimals = priceFeedDecimals;
+    this.periods = periods;
+
+    this.toBN = this.web3.utils.toBN;
+
+    this.convertPriceFeedDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number.
+      // Note: Must ensure that `number` has no more decimal places than `priceFeedDecimals`.
+      return this.toBN(parseFixed(number.toString().substring(0, priceFeedDecimals), priceFeedDecimals).toString());
+    };
+  }
+
+  getCurrentPrice() {
+    return this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
```

This should be an `async` method now and should throw errors instead of returning `null` where a price can't be retrieved. Follow [here](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js#L69) for reference please